src/odemis/acq/stream/_sync.py

```diff
@@ -3142,13 +3143,41 @@ def _runAcquisition(self, future) -> Tuple[List[model.DataArray], Optional[Excep
         :param future: Current future running for the whole acquisition.
         """
         try:
+            # Compute the max safe intensity value, based on the exposure time
+            if self._integrationTime:
+                # calculate exposure time to be set on detector
+                exp = self._integrationTime.value / self._integrationCounts.value  # get the exp time from stream
+                exp = self._ccd.exposureTime.clip(exp)  # set the exp time on the HW VA
+            else:
+                # stream has exposure time
+                exp = self._sccd._getDetectorVA("exposureTime").value  # s
+
+            # Clip to the maximum value of the detector, otherwise we might never detect high intensity
+            # for long exposure times. Should be defined on the streak-ccd.metadata[MD_CALIB]["intensity_limit"]
+            # as count/s.
+            max_det_value = self._ccd.shape[2] - 1
+            ccd_md = self._ccd.getMetadata()
+
+            intensity_limit_s = ccd_md.get(model.MD_CALIB, {}).get("intensity_limit", 40000)
```

Comment:
why is this called `intensity_limit_s` if it is supposed to be specified in counts/s?

Reply:
Because it's "[counts per] s[econd]" . => Will change to  `intensity_limit_cps` & `intensity_limit_cpf`.